pkg/operator/controllers/node/node_controller_test.go

```diff
@@ -0,0 +1,353 @@
+		},
+		{
+			name: "isDraining false, state is degraded but annotationReason is incorrect",
+			node: &corev1.Node{
+				Status: corev1.NodeStatus{
+					Conditions: []corev1.NodeCondition{
+						{
+							Type:   corev1.NodeReady,
+							Status: corev1.ConditionTrue,
+						},
+					},
+				},
+				Spec: corev1.NodeSpec{
+					Unschedulable: true,
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{
+						annotationState:         stateDegraded,
+						annotationReason:        "some-random-reason",
+						annotationCurrentConfig: "foo",
+						annotationDesiredConfig: "bar",
+					},
+				},
+			},
+			want: false,
+		},
+		{
+			name: "isDraining false, annotationReason is correct but state is incorrect",
+			node: &corev1.Node{
+				Status: corev1.NodeStatus{
+					Conditions: []corev1.NodeCondition{
+						{
+							Type:   corev1.NodeReady,
+							Status: corev1.ConditionTrue,
+						},
+					},
+				},
+				Spec: corev1.NodeSpec{
+					Unschedulable: true,
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{
+						annotationState:         "not-a-valid-state",
+						annotationReason:        "failed to drain node",
+						annotationCurrentConfig: "foo",
+						annotationDesiredConfig: "bar",
+					},
+				},
+			},
+			want: false,
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := isDraining(tt.node); tt.want != got {
+				t.Fatalf("error with test '%v': got %v, wanted %v", tt.name, got, tt.want)
+			}
+		})
+	}
+}
+
+func TestGetAnnotation(t *testing.T) {
+	for _, tt := range []struct {
+		name            string
+		objectMeta      *metav1.ObjectMeta
+		annotationKey   string
+		annotationValue string
+	}{
+		{
+			name:            "no annotations set, return a blank annotation",
+			objectMeta:      &metav1.ObjectMeta{},
+			annotationValue: "",
+		},
+		{
+			name: "annotation is set, return its value",
+			objectMeta: &metav1.ObjectMeta{
+				Annotations: map[string]string{"some-random-annotation": "is-set"},
+			},
+			annotationKey:   "some-random-annotation",
+			annotationValue: "is-set",
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := getAnnotation(tt.objectMeta, tt.annotationKey); tt.annotationValue != got {
+				t.Fatalf("error with test '%v': got %v, wanted %v", tt.name, got, tt.annotationValue)
+			}
+		})
+	}
+}
+
+func TestSetAnnotation(t *testing.T) {
+	for _, tt := range []struct {
+		name            string
+		node            *corev1.Node
+		annotationKey   string
+		annotationValue string
+	}{
+		{
+			name:            "no annotations set, set blank ones",
+			node:            &corev1.Node{},
+			annotationKey:   "",
+			annotationValue: "",
+		},
+		{
+			name: "annotations are provided, set them",
+			node: &corev1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{"some-random-annotation": "is-set"},
+				},
+			},
+			annotationKey:   "some-random-annotation",
+			annotationValue: "is-set",
+		},
```

Comment:
What ware we testing in this case? Try intentionally break the function you are testing to make sure that the test case is useful. Will this test case fail if we change `setAnnotation` to a no-op function like this?

```
func setAnnotation(m *metav1.ObjectMeta, k, v string) {
}
```

Reply:
After playing with it a bit, this test wasn't necessary. I have removed it.